scripts/gen_vimdoc.py

```diff
@@ -835,33 +886,156 @@ def is_program_listing(para):
 
     return len(children) == 1 and children[0].nodeName == 'programlisting'
 
-def fmt_node_as_vimhelp(parent: Element, width=text_width - indentation, indent='',
-                        fmt_vimhelp=False):
+
+FunctionParam = Tuple[
+    str,  # type
+    str,  # parameter name
+]
+
+@dataclasses.dataclass
+class FunctionDoc:
+    """Data structure for function documentation. Also exported as msgpack."""
+
+    annotations: List[str]
+    """Attributes, e.g., FUNC_API_REMOTE_ONLY. See annotation_map"""
+
+    notes: List[Docstring]
+    """Notes: (@note strings)"""
+
+    signature: str
+    """Function signature with *tags*."""
+
+    parameters: List[FunctionParam]
+    """Parameters: (type, name)"""
+
+    parameters_doc: Dict[str, Docstring]
+    """Parameters documentation. Key is parameter name, value is doc."""
+
+    doc: List[Docstring]
+    """Main description for the function. Separated by paragraph."""
+
+    return_: List[Docstring]
+    """Return:, or Return (multiple): (@return strings)"""
+
+    seealso: List[Docstring]
+    """See also: (@see strings)"""
+
+    xrefs: List[Docstring]
+    """XRefs. Currently only used to track Deprecated functions."""
+
+    # for INCLUDE_C_DECL
+    c_decl: str | None = None
```

Comment:
Is this syntax supported in Python 3.7? I think you may need to use `Union`

https://docs.python.org/3.7/library/typing.html#typing.Union

Reply:
This is [PEP-563](https://peps.python.org/pep-0563), see `
from __future__ import annotations`. Although the [PEP-604](https://peps.python.org/pep-0604/) syntax is 3.10+ only, PEP-563 allows writing more concise type annotations (which I prefer than Union) for type annotations that are evaluated on TYPE_CHECKING only.

Reply:
Interesting, thanks for the links!

Does PEP-563 also enable us to use e.g. `list`, `dict`, `tuple`, etc. over `List`, `Dict`, `Tuple`?

Reply:
Yes, right. (But somehow I didn't use them in this PR :P)